sound/soc/bcm/hifiberry_dacplus.c

```diff
@@ -406,29 +406,34 @@ static int snd_rpi_hifiberry_dacplus_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return ret;
 	if (ret) {
-		card->aux_dev = hifiberry_dacplus_aux_devs;
-		card->num_aux_devs =
-				ARRAY_SIZE(hifiberry_dacplus_aux_devs);
 		tpa_node = of_find_compatible_node(NULL, NULL, "ti,tpa6130a2");
 		tpa_prop = of_find_property(tpa_node, "status", &len);
-
-		if (strcmp((char *)tpa_prop->value, "okay")) {
-			/* and activate headphone using change_sets */
-			dev_info(&pdev->dev, "activating headphone amplifier");
-			of_changeset_init(&ocs);
-			ret = of_changeset_update_property(&ocs, tpa_node,
-							&tpa_enable_prop);
-			if (ret) {
-				dev_err(&pdev->dev,
-				"cannot activate headphone amplifier\n");
-				return -ENODEV;
-			}
-			ret = of_changeset_apply(&ocs);
-			if (ret) {
-				dev_err(&pdev->dev,
-				"cannot activate headphone amplifier\n");
-				return -ENODEV;
+		if (tpa_prop) {
+			card->aux_dev = hifiberry_dacplus_aux_devs;
+			card->num_aux_devs =
+				ARRAY_SIZE(hifiberry_dacplus_aux_devs);
+			if (strcmp((char *)tpa_prop->value, "okay")) {
```

Comment:
Strictly speaking, the lack of a `status` property also causes a node to be enabled, but see above comment about a better way.